src/transformers/pipelines/text_to_speech.py

```diff
@@ -0,0 +1,173 @@
+
+            return {"input_ids": inputs["input_ids"], "speaker_embeddings": speaker_embeddings}
+
+        processor_args_dict = {}
+
+        SPEAKER_EMBEDDINGS_KEY_MAPPING.get(self.model_type, "speaker_embeddings")
+        if self.model.config.model_type == "bark":
+            # bark speaker embeddings is passed as voice_preset in its processor
+            processor_args_dict["voice_preset"] = speaker_embeddings
+        else:
+            processor_args_dict["speaker_embeddings"] = speaker_embeddings
+
+        output = self.processor(text, **processor_args_dict, **kwargs)
+
+        return output
+
+    def _forward(self, model_inputs, **kwargs):
+        if self.is_speecht5:
+            inputs = model_inputs["input_ids"]
+
+            speaker_embeddings = model_inputs["speaker_embeddings"]
+
+            with torch.no_grad():
+                speech = self.model.generate_speech(inputs, speaker_embeddings, vocoder=self.vocoder)
+        else:
+            if self.only_one_speaker_embeddings:
+                speaker_embeddings_key = SPEAKER_EMBEDDINGS_KEY_MAPPING.get(self.model_type, "speaker_embeddings")
+
+                # check batch_size > 1
+                if len(model_inputs["input_ids"]) > 1 and model_inputs.get(speaker_embeddings_key, None) is not None:
+                    model_inputs[speaker_embeddings_key] = model_inputs[speaker_embeddings_key][0]
+
+            with torch.no_grad():
+                speech = self.model.generate(**model_inputs, **kwargs)
+
+        return speech
+
+    def __call__(
+        self,
+        input_texts: Union[str, List[str]],
+        **generate_kwargs,
+    ):
+        """
+        Generates speech/audio from the inputs. See the [`TextToSpeechPipeline`] documentation for more information.
+
+        Args:
+            input_texts (`str` or `List[str]`):
+                The text(s) to generate.
+            speaker_embeddings (`str` or `torch.Tensor` or `Dict[np.ndarray]`, *optional*):
+                The speaker prompt, i.e the speaker embeddings conditionning the inputs.
+            generate_kwargs (*optional*):
+                Remaining parameters passed to the model generation method.
+
+        Return:
+            A `torch.Tensor` or a list of `torch.Tensor`: Each result comes as a `torch.Tensor` corresponding to the
+            generated audio.
+        """
+        return super().__call__(input_texts, **generate_kwargs)
+
+    def _sanitize_parameters(
+        self,
+        **generate_kwargs,
+    ):
+        preprocess_params = {}
+
+        preprocess_params["speaker_embeddings"] = generate_kwargs.get("speaker_embeddings", None)
+
+        forward_params = {key: val for (key, val) in generate_kwargs.items() if key not in preprocess_params}
+        postprocess_params = {}
+
+        return preprocess_params, forward_params, postprocess_params
+
+    def postprocess(self, speech):
+        return speech
```

Comment:
Pipelines should return an object useable by non ML people.

I don't know what this object is from this code, but I'm pretty sure it's a tensor with no further information.

We can return a numpy array, but we need some way to communicate to the user how is the audio encoded so that he can save it as a file for instance, so f32le, sampling rate at least.

Reply:
I actually added a `self.sampling_rate` argument to `__init__.py` to allow the user to call `pipe.sampling_rate` if ever needed to generate or save audio!

I'm not sure about other ways to do it, do you have ideas?

(BTW, I will modify it to return np array !)